pylint/checkers/base.py

```diff
@@ -384,7 +384,8 @@ def visit_starred(self, node):
                           'duplicate-argument-name', 'nonlocal-and-global')
     def visit_functiondef(self, node):
         self._check_nonlocal_and_global(node)
-        if not redefined_by_decorator(node):
+        if (not redefined_by_decorator(node) and
```

Comment:
this likely should be merged to one function, but I do not have a good idea on how it should be named